dependencies.yaml

```diff
@@ -661,6 +661,10 @@ dependencies:
         packages:
           - dask-cuda==24.4.*
           - *numba
+          - pip
+          - pip:
+              # This should eventually move to rapids-dask-dependency
+              - git+https://github.com/dask-contrib/dask-expr.git@b588a9e15e90e0567061664ffc01374786686e20
```

Comment:
@charlesbluca - Sorry for the ping, but I thought you might have advice :)

I'd like to start running a subset of dask_cudf tests with dask_expr installed (and "enabled" within `dask.dataframe`).  Do you know what kind of hoops I need to jump through to add a new dependency for dask_cudf testing these days?

Reply:
This is a good start, but IIUC think we'll want to split this dependency out because the way it would be specified between conda env files and requirements.txt / pyproject.toml is different; think something like this should work:

```yaml
  test_python_dask_cudf:
    common:
      - output_types: [conda, requirements, pyproject]
        packages:
          - dask-cuda==24.4.*
          - *numba
      - output_types: conda
        packages:
          - pip
          - pip:
              # This should eventually move to rapids-dask-dependency
              - &dask_expr_tip git+https://github.com/dask-contrib/dask-expr.git@b588a9e15e90e0567061664ffc01374786686e20
      - output_types: [requirements, pyproject]
        packages:
          - *dask_expr_tip
```

Essentially, this is just saying that the conda file should include `pip` and the `pip: [git+...]` mapping, while the pyproject and requirements files should only have the `git+...` link itself.

Reply:
Aha. That makes a lot of sense - Thanks @charlesbluca !